src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RandomNumberGenerator.cs

```diff
@@ -165,6 +166,199 @@ public static byte[] GetBytes(int count)
+        /// </summary>
+        /// <param name="choices">The items to use to populate the array.</param>
+        /// <param name="length">The length of array to return populated with items.</param>
+        /// <returns>An array populated with random choices.</returns>
+        /// <typeparam name="T">The type of items.</typeparam>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="choices" /> is empty.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="length" /> is not zero or a positive number.
+        /// </exception>
+        /// <seealso cref="GetString" />
+        /// <seealso cref="GetHexString(Span{char}, bool)" />
+        public static T[] GetItems<T>(ReadOnlySpan<T> choices, int length)
+        {
+            if (choices.IsEmpty)
+                throw new ArgumentException(SR.Arg_EmptySpan, nameof(choices));
+
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            T[] result = new T[length];
+            GetItemsCore<T>(choices, result);
+            return result;
+        }
+
+        /// <summary>
+        ///   Creates a string populated with characters chosen at random from choices.
+        /// </summary>
+        /// <param name="choices">The characters to use to populate the string.</param>
+        /// <param name="length">The length of string to return.</param>
+        /// <returns>A string populated with random choices.</returns>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="choices" /> is empty.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="length" /> is not zero or a positive number.
+        /// </exception>
+        /// <seealso cref="GetItems{T}(ReadOnlySpan{T}, Span{T})" />
+        /// <seealso cref="GetItems{T}(ReadOnlySpan{T}, int)" />
+        /// <seealso cref="GetHexString(Span{char}, bool)" />
+        public static unsafe string GetString(ReadOnlySpan<char> choices, int length)
+        {
+            if (choices.IsEmpty)
+                throw new ArgumentException(SR.Arg_EmptySpan, nameof(choices));
+
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            fixed (char* pChoices = choices)
+            {
+                return string.Create(length,
+                    (ChoicesPointer: (IntPtr)pChoices, ChoicesLength: choices.Length),
+                    static (destination, state) =>
+                    {
+                        ReadOnlySpan<char> choices = new ReadOnlySpan<char>(state.ChoicesPointer.ToPointer(), state.ChoicesLength);
+                        GetItemsCore(choices, destination);
+                    });
+            }
```

Comment:
```suggestion
            return string.Create(length,
                (IntPtr)&choices,
                static (destination, state) =>
                {
                    GetItemsCore(*(ReadOnlySpan<char>*)state, destination);
                });
```
This is valid now in C#11 I think.

Reply:
Is this safe?
`choices` could be given as `char[]` which can be relocated by the GC. 
Imagine: the address is taken, then the GC compacts and moves choices to different address -- that change isn't reflected in the taken address, so at dereferencing it's at invalid address.
So `choices` must be pinned to be safe.

Reply:
`choices` can be moved by the GC, but _`&choices`_ will be not, because it is in the stack.

Reply:
> Is this safe?

Yes. The GC would be moving the reference within the `Span<T>`, but not the `Span<T>` itself. What is being passed here is the address of the local containing the `Span<T>`, not the reference within it.

Reply:
Thanks.
My mental struggle was that pinning the span is actually pinning the underlying ref (which I somehow didn't think about (on Sunday)).

Reply:
For me this results in a compilation error.

> This takes the address of, gets the size of, or declares a pointer to a managed type ('ReadOnlySpan\<char\>')

We might be waiting on a Roslyn bump for it. This pattern appears in a number of places so it would be good to clean them all up when the feature is available.

Reply:
@vcsjones this is a warning that gets treated as an error. We can suppress it (and perhaps globally).

Reply:
Ah. My tooling is out of date so I was seeing CS0208.

> (and perhaps globally)

This warning already has targeted suppressions, so I'm inclined to follow the pattern that already exists. Globally supressing a warning is above my pay grade ($0).